.github/workflows/ci.yaml

```diff
@@ -41,3 +41,7 @@ jobs:
         run: make build-c
       - name: "Run Unit Tests"
         run: make test
+        env:
```

Comment:
As the comment above, I don't think it's a good idea to run unit tests that read environment variables from the host machine. I will give another reason why this is not ideal.

Imagine if someone cloned our repo, and suddenly had no Internet access. At least they can do local development and run unittests to test their code. If you add external provider connections, then there is no way unittests will pass if there is no Internet connection.

Reply:
I've moved this to integration tests ✅